src/Core/Grid/Position/GridPositionMultiStoreUpdater.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Position;
+
+use PrestaShop\PrestaShop\Core\Grid\Position\UpdateHandler\PositionUpdateHandlerInterface;
+
+/**
+ * Class GridPositionMultiStoreUpdater, this class is responsible for updating the position of items
+ * of a grid using the information from a PositionUpdateInterface object but with respecting multi-store.
+ */
+final class GridPositionMultiStoreUpdater implements GridPositionUpdaterInterface
+{
+    /**
+     * @var PositionUpdateHandlerInterface
+     */
+    private $updateHandler;
+
+    /**
+     * @param PositionUpdateHandlerInterface $updateHandler
+     */
+    public function __construct(PositionUpdateHandlerInterface $updateHandler)
+    {
+        $this->updateHandler = $updateHandler;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update(PositionUpdateInterface $positionUpdate)
+    {
+        $newPositions = $this->getNewPositions($positionUpdate);
+        $this->sortByPositionValue($newPositions);
+        $this->updateHandler->updatePositions($positionUpdate->getPositionDefinition(), $newPositions);
+    }
+
+    /**
+     * @param PositionUpdateInterface $positionUpdate
+     *
+     * @return array
+     */
+    private function getNewPositions(PositionUpdateInterface $positionUpdate)
+    {
+        $positions = $this->updateHandler->getCurrentPositions($positionUpdate->getPositionDefinition(), $positionUpdate->getParentId());
+
+        /** @var PositionModificationInterface $rowModification */
+        foreach ($positionUpdate->getPositionModificationCollection() as $rowModification) {
+            $positions[$rowModification->getId()] = $rowModification->getNewPosition();
+        }
+
+        return $positions;
+    }
+
+    /**
+     * @param array $positions
+     */
+    private function sortByPositionValue(&$positions)
+    {
+        asort($positions);
```

Comment:
That's one of the "clean code" precepts, a well-named function is better than having to write a comment explaining what is happening.

It could be considered overkill, but I kinda like it too...

Reply:
It's overkill, you're not creating method each time you have to use `empty($array)` with name `checkIfArrayIsEmpty`.
I'm sad that a senior developer like him doesn't know the sort function which is a very old PHP4 function :(

Reply:
it's in between...  the name of the variable `$newPositions` is self-explanatory, we could remove the comment and not use a function, either way, it is fine to me

Reply:
(this is the epitome of knit-picking though :D )